Assets/MixedRealityToolkit.Extensions/Webrtc/Scripts/WebrtcPeerEvents.cs

```diff
@@ -0,0 +1,389 @@
+        /// </summary>
+        [Tooltip("This will log all errors to the Unity console")]
+        public bool AutoLogErrors = true;
+
+        /// <summary>
+        /// Event that occurs when the peer is ready
+        /// </summary>
+        /// <remarks>
+        /// This is the same as <see cref="Webrtc.OnInitialized"/> but is included here for convienience
+        /// </remarks>
+        [Tooltip("Event that occurs when the peer is ready")]
+        public UnityEvent OnPeerReady = new UnityEvent();
+
+        /// <summary>
+        /// Event that occurs when data is ready from the data channel
+        /// </summary>
+        [Tooltip("Event that occurs when data is ready from the data channel")]
+        public WebrtcPeerStringEvent OnDataFromDataChannelReady = new WebrtcPeerStringEvent();
+
+        /// <summary>
+        /// Event that occurs when a peer-specific failure happens
+        /// </summary>
+        [Tooltip("Event that occurs when a peer-specific failure happens")]
+        public WebrtcPeerStringEvent OnFailure = new WebrtcPeerStringEvent();
+
+        /// <summary>
+        /// Event that occurs when an ice candidate is prepared
+        /// </summary>
+        [Tooltip("Event that occurs when an ice candidate is prepared")]
+        public WebrtcPeerStringIntStringEvent OnIceCandiateReadyToSend = new WebrtcPeerStringIntStringEvent();
+
+        /// <summary>
+        /// Event that occurs when the data channel is ready
+        /// </summary>
+        [Tooltip("Event that occurs when the data channel is ready")]
+        public UnityEvent OnLocalDataChannelReady = new UnityEvent();
+
+        /// <summary>
+        /// Event that occurs when an sdp message is prepared
+        /// </summary>
+        [Tooltip("Event that occurs when an sdp message is prepared")]
+        public WebrtcPeerString2Event OnLocalSdpReadyToSend = new WebrtcPeerString2Event();
+
+        /// <summary>
+        /// Event that occurs when an sdp offer is prepared
+        /// </summary>
+        [Tooltip("Event that occurs when an sdp offer is prepared")]
+        public WebrtcPeerStringEvent OnSdpOfferReadyToSend = new WebrtcPeerStringEvent();
+
+        /// <summary>
+        /// Event that occurs when an sdp answer is prepared
+        /// </summary>
+        [Tooltip("Event that occurs when an sdp answer is prepared")]
+        public WebrtcPeerStringEvent OnSdpAnswerReadyToSend = new WebrtcPeerStringEvent();
+
+        /// <summary>
+        /// The underlying peer
+        /// </summary>
+        private IPeer peer = null;
+
+        /// <summary>
+        /// Internal queue used to marshal work back to the main unity thread
+        /// </summary>
+        private Queue<Action> MainThreadWorkloads = new Queue<Action>();
+
+        /// <summary>
+        /// Initialization handler
+        /// 
+        /// TODO(bengreenier): This could be auto-bound with editor script
+        /// </summary>
```

Comment:
nit: don't check in todos. Open a separate issue for this todo or remove the code/call out what is not implemented with no intention of fixing.